site/gatsby-site/src/utils/AlgoliaUpdater.js

```diff
@@ -46,30 +46,37 @@ class AlgoliaUpdater {
     this.algoliaClient = algoliaClient;
   }
 
-  generateIndexEntries({ reports, classifications }) {
+  generateIndexEntries = async ({ reports, incidents, classifications }) => {
     let classificationsHash = {};
 
-    classifications.map((c) => {
+    classifications.forEach((c) => {
       classificationsHash[c.incident_id] = getClassificationArray(c.classifications, c.namespace);
     });
 
-    const downloadData = reports.map((report) => {
-      const finalDataNode = {
-        objectID: report.report_number.toString(),
-        ...report,
-      };
+    const downloadData = [];
 
-      if (classificationsHash[report.incident_id]) {
-        finalDataNode.classifications = classificationsHash[report.incident_id];
-      }
+    for (const incident of incidents) {
+      for (const report_number of incident.reports) {
+        const report = reports.find((r) => r.report_number == report_number);
```

Comment:
This looks a bit wasteful. I think it may be worth opening an issue for it so we can land this.